examples/main.tf

```diff
@@ -4,31 +4,19 @@
 module "vcn" {
   source = "../"
 
-  region = "us-phoenix-1"
-
   # general oci parameters
-
-  compartment_id = ""
-
-  label_prefix = "dev"
+  compartment_id = var.tenancy_id
```

Comment:
The logic behind this was that `tenancy_id` is always set in the process to authenticate to the OCI API, while `compartment_id` is not necessarily set and we cannot determine a value ahead of time.

The only known value for compartment regardless of the environment is the tenancy ocid :-(

I agree creating any resource in the root compartment of a tenancy should be generally avoided, but that's ultimately the module user's choice and responsibility.

If the goal of the example is to have a resource deployed with the least amount of friction, I would prefer to have a copy/paste experience that works with minimum effort, and then tweak it to my needs (e.g. provide a specific compartment). Also `compartment_id` is updatable, so user can adjust the final result if needed.

Would it be a better solution if we write the example code in a way that works out of the box, and add a comment to advise setting something more relevant values for user's context?

```suggestion
  compartment_id = var.tenancy_id # please set `var.compartment_id` value and change `var.tenancy_id` to `var.compartment_id` to avoid creating resources in the root compartment
```

Reply:
I don't agree with this. We use the tenancy_id mainly for initializing providers whereas compartment_id to create resources. 

Yes, technically root compartment is the tenancy id but and I know it is possible to create stuff in root compartment but that's promoting bad practices and we shouldn't be doing that.  Users should explicitly enter compartment_id of where they want their resources created. If they don't root = tenancy ids but the need to pass a compartment id forces them to create one anyway, then we are guiding them towards good practices.

Reply:
OK. I guess this decision will make his way into the codingconventions.adoc 👍